### DISCUSSION TITLE
How to create a connecting line between two asymmetrical shapes in fabric.js?

### DISCUSSION THREAD

**Discussion by chhavikohli** [fabricjs/fabric.js#8732] [Q&A]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

5.0.0

### Node Version (if applicable)

None

### Link To Reproduction

https://jsfiddle.net/2btqharv/6/

### Steps To Reproduce

1. Create two triangles.
2. Create a line between the triangles.

### Expected Behavior

REQUIREMENT:- Let's take an example we have two triangles or any other shape and I want two connect them with a line - enter image description here
<img width="858" alt="Screenshot 2023-02-23 at 12 28 04 PM" src="https://user-images.githubusercontent.com/31480080/220843907-745441dd-5b04-4216-ae5e-805c2886236b.png">

PROBLEM:- I am getting boundingRect coordinates and line is not drawn properly.The line in the below image is drawn from MR control of first object to ML control of second object and not touching the vertex of the shape.enter image description here.
![download](https://user-images.githubusercontent.com/31480080/220843938-666ea291-6405-413d-9860-d4a9ec93a106.png)

I want to get coordinates of the shape namely (x1,y1),(x2,y2),(x3,y3),(x4,y4) in the first image. Assuming that after fetching these points I can use calculations to find my target points. Any other solution or ideas will be highly appreciated.
<img width="433" alt="Screenshot 2023-02-23 at 12 21 48 PM" src="https://user-images.githubusercontent.com/31480080/220843961-daedae29-76e5-4093-8f28-f0b199591861.png">

FABRIC VERSION- Fabric 5

const fc = new fabric.Canvas("c");

const triangle1Options = {
    stroke: 'black',
    strokeWidth: 2,
    fill: 'red',
    left: 10,
    top: 10,
    width: 100,
    height: 100,
}

let triangle1 = new fabric.Triangle(triangle1Options);

fc.add(triangle1);

// Get coords of triangle
/* The below syntax returns => {
  height: 102,
  left: 10,
  top: 10,
  width: 102
}
This is the bounding rect of triangle. How to get actual coordinates of the triangle. */
fc.on('selection:created', event => {console.log(event.target.getBoundingRect())});

[Demo](https://jsfiddle.net/2btqharv/6/)

### Actual Behavior

![download](https://user-images.githubusercontent.com/31480080/220844009-76ca6496-27d1-4fc7-959a-63afc51fa49f.png)

**Comment by DineshCodeFlow** [comment]
I am also trying to find the actual object coordinates and explored the docs too. Couldn't find any method
I hope you find an answer soon

**Comment by ShaMan123** [comment]
I suggest you look into poly controls.
That will give you the coordinates.
Then use events/brush to add/draw the line
https://github.com/fabricjs/fabric.js/blob/8bfb7bd9500d0278ef27280e7360071e6b4d6bb6/src/controls/polyControl.ts#L24-L35

**Reply by chhavikohli** [reply]
Thank you @ShaMan123 for checking out the issue I'm facing.
I updated my fiddle https://jsfiddle.net/z2chpmas/13/ with polygon

```
var poly = new fabric.Polygon( [
    { x: 100, y: 50 },
    { x: 200, y: 150 },
    { x: 100, y: 250 }
  ], {
  fill: 'blue',
  //positionHandler: fabric.factoryPolyPositionHandler(poly)
});
fc.add(poly);
```
without `factoryPolyPositionHandler` I am able to identify the mid-points.

I have one question-

 `fabric.Polygon` accepts the points, I tried n-number of solutions to convert path to points, but unable to find anything accurate. Do you have something that could make this thing easy?

Thank you so much again!

**Reply by ShaMan123** [reply]
`fabric.controlsUtils.factoryPolyPositionHandler(poly)` is part of v6, not sure it is exported though.
jsfiddle can't be used until we publish a version to cdn
read the source code for usage, it is straight forward
